### ISSUE TITLE
Backend CI testing with GitHub Actions

### ISSUE THREAD

**Issue by DominicLuidold** [DominicLuidold/project-paperless#16]
**User story**
As a `backend developer`, I `want to have automated tests whenever a pull request is created`, `so that I can verify that my changes work by inspecting the CI results`.

**Description of the task**
Setup GitHub Actions workflows that builds a custom PHP Ci image based on the backend [Dockerfile](https://github.com/DominicLuidold/project-paperless/blob/da2d3d6e9e494f32901dd39fac7fbf730d38ad63/docker/backend/Dockerfile) and uses that to run tests, check code style, and security.

**Specifications**
- Build workflow uses
  - Backend Dockerfile
  - https://ghcr.io for publishing the Docker image
  - Runs nighly builds
- Test workflow uses CI image, runs
  - PHP-CS-Fixer
  - PHPStan
  - PHPUnit
  - Security checks
    - Composer
    - Symfony
- Caching is used where possible
- Workflows run for every pull request and when pushing to `main` (`stage` and `production` later on)
- Successful runs are required for a pull request to be merged

**Acceptance criteria**
- [ ] CI workflows are triggered when opening/pushing commits to a pull request or when pushing to `main`
  - [ ] Workflows are required for pull requests
- [ ] Individual jobs are configured and reports available for PHP-CS-Fixer, PHPStan etc.

**Out of scope**
- CI images per branch
- CI testing for frontend/Angular (depends on #11)
- Automated deployments

**Additional context**
/
